campaigns/tests.py

```
@@ -0,0 +1,317 @@
+        self.assertEqual(donations[0], donation2)  # Newer donation first
+        self.assertEqual(donations[1], donation1)
+
+    def test_unique_constraint(self):
+        """Test unique constraint on (on_chain_id, campaign)"""
+        CampaignDonation.objects.create(**self.donation_data)
+
+        # Try to create another donation with same on_chain_id and campaign
+        donation_data_2 = self.donation_data.copy()
+        donation_data_2["total_amount"] = "500000000000000000000000"
+
+        with self.assertRaises(Exception):  # Should raise IntegrityError
```

Ruff (0.12.2)

Do not assert blind exception: `Exception`

(B017)